internal/controller/ai_gateway_route.go

```diff
@@ -43,6 +43,12 @@ const (
 	egOwningGatewayNamespaceLabel                      = egAnnotationPrefix + "owning-gateway-namespace"
 	// apiKeyInSecret is the key to store OpenAI API key.
 	apiKeyInSecret = "apiKey"
+	// GatewayConfigAnnotationKey is the annotation key used on Gateway objects to reference a GatewayConfig.
+	// The value should be the name of the GatewayConfig resource in the same namespace as the Gateway.
+	GatewayConfigAnnotationKey = "aigateway.envoyproxy.io/gateway-config"
+	// GatewayConfigFinalizerName is the finalizer added to GatewayConfig resources to prevent deletion
+	// while they are still referenced by Gateway objects.
+	GatewayConfigFinalizerName = "aigateway.envoyproxy.io/gateway-config-protection"
```

Comment:
we don't have anything like this for other resources, so let's remove the finalizer stuff for now, and think about this one later after this PR together withe other resources